src/dawn/IIR/StencilInstantiation.cpp

```diff
@@ -47,7 +47,9 @@ namespace iir {
 //===------------------------------------------------------------------------------------------===//
 
 StencilInstantiation::StencilInstantiation(dawn::OptimizerContext* context)
-    : context_(context), IIR_(make_unique<IIR>()), SIR_(context->getSIR()) {}
+    : context_(context), metadata_(*(context->getSIR()->GlobalVariableMap)),
```

Comment:
it is store only in IIR, but we need it in order to construct certain metadata of the StencilMetaInformation. Therefore it is passed to the ctr that will build that metadata (instead of constructing empty containers and initializing later, which leaves the object in inconsistent state after ctr) but not stored.

Reply:
Thanks, this is clear now.